Docs: Update doc pages to include `curvature_on` kwarg (#253)
MDAnalysis/membrane-curvature: docs/source/pages/Algorithm.rst

:attr:`MembraneCurvature.results.average_gaussian` arrays, respectively.
Each array has shape ``(n_x_bins, n_y_bins)``.

How those averages are built depends on ``curvature_on`` and ``fft_filter``:

> - With ``curvature_on='per_frame'``, average maps are time averages of per-frame curvature maps:

Reviewer: ```suggestion
- With ``curvature_on='per_frame'``, the average curvature maps are the time average of the per-frame curvatures:
```

  :math:`\langle H \rangle` and :math:`\langle K \rangle`.

> - With ``curvature_on='average_surface'``, average maps are curvatures evaluated directly on the time-averaged surface height field:

Reviewer: ```suggestion
- With ``curvature_on='average_surface'``, the average curvature maps are calculated directly on the time-averaged surface height field:
```

  :math:`H(\langle S \rangle)` and :math:`K(\langle S \rangle)`.

> - Leaving ``curvature_on=None`` (default) preserves prior behavior: ``'per_frame'`` when ``fft_filter=None``, and ``'average_surface'`` when ``fft_filter`` is set.

Reviewer: This style is more appropriate for API docs, where the reader can easily navigate and identify this is new in a given version added.

Rephrase this as text with no reference to prior behaviour.

4.1 Distinguishing curvature of average surface vs. average of per-frame curvatures
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Because curvature is a non-linear function of the surface height field :math:`z(x, y)`,

> evaluating curvature on the time-averaged surface is mathematically distinct from time-averaging per-frame curvature maps:

Reviewer: Usually for documentation pages, language is more natural than in API docs. Even if this is an algorithm page, the idea is to communicate what the code does, but in a languague that is accessible to diverse audiences.

```suggestion
deriving curvature from a time-averaged surface is different from calculating curvature by time averaging curvature maps on every frame:
```

.. math::

   \langle H \rangle \neq H(\langle S \rangle), \qquad \langle K \rangle \neq K(\langle S \rangle)

where:

- :math:`\langle H \rangle` and :math:`\langle K \rangle` are the time-averaged per-frame curvatures (the mean over all frames of per-frame curvature maps :math:`H_i` and :math:`K_i`).

> - :math:`H(\langle S \rangle)` and :math:`K(\langle S \rangle)` are the mean and Gaussian curvatures evaluated directly on the time-averaged surface height field :math:`\langle S \rangle = \text{average\_z\_surface}` (including after optional FFT filtering).

Reviewer: These two bullet points can be simplified to one or two lines that continue the `where` in line 679. We can also avoid the math syntax. Just plain language to explain. 

Bullet points are good, but usage shouls be limited since it breaks the continuity and flow of the text. Also, since we don't use the subindex `i`, we can skip all of that.

The two approaches capture distinct physical quantities:

> - **Per-frame average** (:math:`\langle H \rangle`, :math:`\langle K \rangle`): Evaluates curvature for each frame individually and then averages over time. This preserves the mean contribution of instantaneous thermal fluctuations.

Reviewer: "Evaluate" is not the best word to describe MembraneCurvature. We either derive (for a specific method, like binning), or compute/calculate.

```suggestion
- **Per-frame average** (:math:`\langle H \rangle`, :math:`\langle K \rangle`): Calculates curvature for each frame individually and then averages over time. This preserves the mean contribution of instantaneous thermal fluctuations.
```

Although "evaluate" can be mathematically correct when there is a function and then we evaluate at a point, what MC does is compute curvature formulas on a z(x,y) field.

> - **Average surface curvature** (:math:`H(\langle S \rangle)`, :math:`K(\langle S \rangle)`): Time-averages the surface height field :math:`z(x,y)` over the trajectory first (allowing uncorrelated thermal fluctuations to cancel out), and then evaluates curvature on the resulting smooth time-averaged surface :math:`\langle S \rangle`.

Reviewer: Since we use a lot of symbols here, rephrase this bullet point to convey about the same information but without the parenthesis. Similar to bullet points, it can disrupt. It's a short idea that can be conveyed to the reader with a more regular flow.

4.2 Controlling average curvature evaluation mode (``curvature_on``)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> The evaluation mode for :attr:`~membrane_curvature.base.MembraneCurvature.results.average_mean` and :attr:`~membrane_curvature.base.MembraneCurvature.results.average_gaussian` is controlled by the ``curvature_on`` parameter:

Reviewer: ```suggestion
The calculation of  :attr:`~membrane_curvature.base.MembraneCurvature.results.average_mean` and :attr:`~membrane_curvature.base.MembraneCurvature.results.average_gaussian` is controlled by the ``curvature_on`` parameter:
```

- ``curvature_on='per_frame'``: Computes :math:`\langle H \rangle` and :math:`\langle K \rangle` as the temporal mean of per-frame curvature maps (``results.mean`` and ``results.gaussian``).
- ``curvature_on='average_surface'``: Evaluates :math:`H(\langle S \rangle)` and :math:`K(\langle S \rangle)` on ``results.average_z_surface``.
- ``curvature_on=None`` (default): Preserves prior default behaviors:
  - When ``fft_filter`` is ``None``, defaults to ``'per_frame'``.

>   - When ``fft_filter`` is enabled (``'auto'`` or a custom passband dict), defaults to ``'average_surface'`` (evaluates curvature on the FFT-filtered time-averaged surface).

Reviewer: We don't need a bullet point for the filter ones, just some text will do:

```suggestion

Note that with the brick-wall FFT filter, curvature is calculated on the time-averaged surface. Without it, the default is a per-frame average.
```

:attr:`MembraneCurvature.results.average_gaussian` arrays, respectively.
Each array has shape ``(n_x_bins, n_y_bins)``.

How those averages are built depends on ``curvature_on`` and ``fft_filter``:

- With ``curvature_on='per_frame'``, the average curvature maps are the time average of the per-frame curvatures.
- With ``curvature_on='average_surface'``, the average curvature maps are calculated directly on the time-averaged surface height field.

When ``curvature_on=None`` (the default), curvature is calculated on the time-averaged surface if an FFT filter is specified, or as a per-frame average if no filter is used.

> 4.1 Distinguishing curvature of average surface vs. average of per-frame curvatures

Reviewer: ```suggestion
4.1 Calculating curvature from average surface vs. average of per-frame curvatures
```

Because curvature is a non-linear function of the surface height field :math:`z(x, y)`,
deriving curvature from a time-averaged surface is different from calculating curvature by time averaging curvature maps on every frame:

.. math::

   \langle H \rangle \neq H(\langle S \rangle), \qquad \langle K \rangle \neq K(\langle S \rangle)

where :math:`\langle H \rangle` and :math:`\langle K \rangle` are the time-averaged per-frame curvatures, while :math:`H(\langle S \rangle)` and :math:`K(\langle S \rangle)` represent the mean and Gaussian curvatures calculated directly on the time-averaged surface height field.

The two approaches capture distinct physical quantities:

- **Per-frame average**: Calculates curvature for each frame individually and then averages over time. This preserves the mean contribution of instantaneous thermal fluctuations.
- **Average surface curvature**: Time-averages the surface height field :math:`z(x,y)` over the trajectory first (allowing uncorrelated thermal fluctuations to cancel out), and then calculates curvature on the resulting smooth time-averaged surface.

> 4.2 Controlling average curvature calculation mode (``curvature_on``)

Reviewer: ```suggestion
4.2 Curvature calculation mode (``curvature_on``)
```
